File: Haskell/FizzBuzz/test_FizzBuzz.hs
module Test_FizzBuzz where

import Test.HUnit
import FizzBuzz

div_by_three_only_gets_fizz_test = TestCase (assertEqual "Test1" "Fizz" (fizzBuzzRep 9))
div_by_five_only_gets_buzz_test = TestCase (assertEqual "Test2" "Buzz" (fizzBuzzRep 25))
div_by_three_and_five_gets_fizzbuzz_test = TestCase (assertEqual "Test3" "FizzBuzz" (fizzBuzzRep 45))
div_by_neither_gets_number_itself_test = TestCase (assertEqual "Test4" "7" (fizzBuzzRep 7))
list_to_fizzbuzz_works_on_small_list_test = TestCase (assertEqual "Test5" ["FizzBuzz", "1", "2", "Fizz", "4", "Buzz"] (listToFizzBuzz [0..5]))


tests = TestList [div_by_three_only_gets_fizz_test,
                  div_by_five_only_gets_buzz_test,
                  div_by_three_and_five_gets_fizzbuzz_test,
                  div_by_neither_gets_number_itself_test,
                  list_to_fizzbuzz_works_on_small_list_test]

-- main = do runTestTT tests

File: Ruby/FizzBuzz/test_fizz_buzz.rb
require './fizz_buzz'
require 'test/unit'

class TestFizzBuzz < Test::Unit::TestCase
  def test_fizz_buzz_rep_of_three_is_fizz
    answer = FizzBuzz.representation 3
    expected = "Fizz"

    assert_equal expected, answer
  end

  def test_fizz_buz_rep_of_ten_is_buzz
    answer = FizzBuzz.representation 10
    expected = "Buzz"

    assert_equal expected, answer
  end

  def test_fizz_buzz_rep_of_seven_is_seven
    answer = FizzBuzz.representation 7
    expected = "7"

    assert_equal expected, answer
  end

  def test_fizz_buzz_range_works_for_a_small_range
    answer = FizzBuzz.range 0, 6
    expected = ["FizzBuzz", "1", "2", "Fizz", "4", "Buzz", "Fizz"]

    assert_equal expected, answer
  end
end
